### PR TITLE
Improved determination of root domain for the Variomedia DNS API

### PR DESCRIPTION
**Pull request by peterkelm**
Rewrite of `_get_root()` to address an issue with the handling of subdomains in the `/domains` call of the Variomedia DNS API. See #2564.

### CODE REVIEW

File: dnsapi/dns_variomedia.sh [hunk 1]
```diff
@@ -93,11 +89,12 @@ dns_variomedia_rm() {
 # _sub_domain=_acme-challenge.www
 # _domain=domain.com
 _get_root() {
-  fulldomain=$1
-  i=1
+  domain=$1
+  i=2
```
Comment (Neilpang) [inline dnsapi/dns_variomedia.sh:93]: the value of "i" should be begging from `1`, not `2`.

In dns alias mode,  the fulldomain can be no  `_acme-challenge`  prefix

Comment (morlac) [reply dnsapi/dns_variomedia.sh:93]: it's been tested:
- successfully with i=2
- unsuccessfully with i=1

**Comment by fraenki** [comment]
@peterkelm Using this patch, I can confirm that it fixes the *creation* of TXT records with acme.sh. However, in my case it failed to *remove* the TXT records afterwards. Some unusual cURL output is shown and then a failure is reported:

```
[Mon Feb 21 00:21:46 CET 2022] Here is the curl dump log:
[Mon Feb 21 00:21:46 CET 2022] == Info:   Trying 81.28.224.32:443...
== Info: Connected to api.variomedia.de (81.28.224.32) port 443 (#0)
== Info: ALPN, offering h2
== Info: ALPN, offering http/1.1
== Info:  CAfile: /usr/local/etc/ssl/cert.pem
== Info:  CApath: none
=> Send SSL data, 5 bytes (0x5)
...
[Mon Feb 21 00:21:46 CET 2022] ret='3'
[Mon Feb 21 00:21:46 CET 2022] Error dns-records?filter[domain]=example.com
[Mon Feb 21 00:21:46 CET 2022] Error
[Mon Feb 21 00:21:46 CET 2022] Error removing txt for domain:_acme-challenge.example.com
```

**Comment by TobiasGrave** [comment]
> @peterkelm Using this patch, I can confirm that it fixes the _creation_ of TXT records with acme.sh. However, in my case it failed to _remove_ the TXT records afterwards. Some unusual cURL output is shown and then a failure is reported:
> 
> ```
> [Mon Feb 21 00:21:46 CET 2022] Here is the curl dump log:
> [Mon Feb 21 00:21:46 CET 2022] == Info:   Trying 81.28.224.32:443...
> == Info: Connected to api.variomedia.de (81.28.224.32) port 443 (#0)
> == Info: ALPN, offering h2
> == Info: ALPN, offering http/1.1
> == Info:  CAfile: /usr/local/etc/ssl/cert.pem
> == Info:  CApath: none
> => Send SSL data, 5 bytes (0x5)
> ...
> [Mon Feb 21 00:21:46 CET 2022] ret='3'
> [Mon Feb 21 00:21:46 CET 2022] Error dns-records?filter[domain]=example.com
> [Mon Feb 21 00:21:46 CET 2022] Error
> [Mon Feb 21 00:21:46 CET 2022] Error removing txt for domain:_acme-challenge.example.com
> ```

The removal of the the TXT record fails when its content starts with a `-`. 

This problem is caused by `grep "$txtvalue"` in the following line:
`record_id="$(echo "$response" | cut -d '[' -f2 | cut -d']' -f1 | sed 's/},[ \t]*{/\},§\{/g' | tr § '\n' | grep "$_sub_domain" | grep "$txtvalue" | sed 's/^{//;s/}[,]?$//' | tr , '\n' | tr -d '\"' | grep ^id | cut -d : -f2 | tr -d ' ')"`

This is due to grep interpreting everything starting with a `-` as an option and not as a search term.
In order to fix this, the option `--` has to be added before the search term (https://www.gnu.org/software/grep/manual/grep.html#index-_002d_002d):
`_record_id="$(echo "$response" | cut -d '[' -f2 | cut -d']' -f1 | sed 's/},[ \t]*{/\},§\{/g' | tr § '\n' | grep "$_sub_domain" | grep -- "$txtvalue" | sed 's/^{//;s/}[,]?$//' | tr , '\n' | tr -d '\"' | grep ^id | cut -d : -f2 | tr -d ' ')"`

**Pull request closed without merging**